### ISSUE TITLE
Under OrdinalModule

### ISSUE THREAD

**Issue by jaideep11061982** [EthanRosenthal/spacecutter#5]
Hi could u help understand the below peace of code. 
Why do we subtract the elements  in linkmat  ,then concatinating them . 
Isnt just cutpoints-X is sufficient ?
```
sigmoids=cutpoints-X
 link_mat = sigmoids[:, 1:] - sigmoids[:, :-1]
        link_mat = torch.cat((
                sigmoids[:, [0]],
                link_mat,
                (1 - sigmoids[:, [-1]])
            ),
            dim=1
```

2) when does this AscensionCallback gets called up.. start of every batch,epoch,or end of batch or epoch

**Comment by EthanRosenthal** [comment]
The `link_mat` comes from the middle line of this equation (from my [blog post](https://www.ethanrosenthal.com/2018/12/06/spacecutter-ordinal-regression/) on this)

![image](https://user-images.githubusercontent.com/7435500/84772577-226b1b80-afa9-11ea-8106-a72c6dc83763.png)

Adjacent cutpoints have to be subtracted from each other.

The `AscenscionCallback` gets [called](https://github.com/EthanRosenthal/spacecutter/blob/master/spacecutter/callbacks.py#L40) at the end of every batch.

**Comment by jaideep11061982** [comment]
NOt sure why but my loss keeps circulating around few values so model dsn converges
```
self.link = LogisticCumulativeLink1(6,
                                           init_cutpoints='ordered')
```
 ```
def forward(self,  x):
      
        x = self.enc(x)
       
        x = self.head(x) # it output dim is 1 
        x=self.link(x)
        
      
        return x
```
`loss=CumulativeLinkLoss()`

**Comment by EthanRosenthal** [comment]
Sorry I can't really help debug -- it could have something to do with the library or be something else. Perhaps you can try training as a regular regression model to see if you can fit that. If you can, then maybe it's something to do with ordinal regression, specifically.
